dali/kernels/signal/fft/fft_cpu_test.cc

```diff
@@ -0,0 +1,468 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <tuple>
+#include <vector>
+#include <complex>
+#include <cmath>
+#include "dali/kernels/scratch.h"
+#include "dali/kernels/signal/fft/fft_cpu.h"
+#include "dali/test/test_tensors.h"
+#include "dali/test/tensor_test_utils.h"
+
+namespace dali {
+namespace kernels {
+namespace signal {
+namespace fft {
+namespace test {
+
+void NaiveDft(std::complex<float> *out,
```

Comment:
Can we change pointer to references to vectors?

Reply:
The idea is that this can be used not only on vectors but also tensors, etc. I could change it to span though

Reply:
changed to std::vector and span where appropriate